test-cases/tests/performance/k04-run-performance-test-against-rhoam.md

```diff
@@ -0,0 +1,98 @@
+
+    Note: This not possible for installations via addon-flow since Hive would revert your modifications to whatever is set in Managed Tenants repository in [sku-limits.yaml.j2](https://gitlab.cee.redhat.com/service/managed-tenants/-/blob/master/addons/managed-api-service/metadata/stage/sku-limits.yaml.j2) file.
+
+7. Run the performance test suite
+
+   The way to do it is described in [MGDAPI-238](https://issues.redhat.com/browse/MGDAPI-238) and in [Austin's doc](https://docs.google.com/document/d/1NJBUsieRkBLnN2PMAF5cpaH7uXq9mZCx1JQaT9Ruytk/edit?usp=sharing). Use [trepel branch](https://gitlab.cee.redhat.com/trepel/3scale-py-testsuite/-/tree/performance_tests). To validate the advertised load use [rhsso_tokens](https://gitlab.cee.redhat.com/trepel/3scale-py-testsuite/-/blob/performance_tests/testsuite/tests/performance/apicast/smoke/template_rhsso_tokens.hf.yaml) benchmark. It is set to have 10% of login flow requests. You will need to change `maxSessions` (~6000), `usersPerSec` (~25 to validate 20M load), `duration`, `maxDuration`, and [http.sharedConnections](https://gitlab.cee.redhat.com/trepel/3scale-py-testsuite/-/blob/performance_tests/testsuite/tests/performance/apicast/smoke/test_smoke_rhsso_tokens.py#L54-55) (~1200).
+
+- create a `perf-test-start-time.txt` file as described in [capture_resource_metrics script](https://github.com/integr8ly/integreatly-operator/blob/master/scripts/capture_resource_metrics.sh).
+  - the actual performance test run doesn't start immediately, the performance test suite creates various 3scale (Product, Backend, Application, Application Plan) and SSO (realm, client, users) entities first
+  - best to track the log of Hyperfoil controller to get the exact time the when `rampUp` phase starts
+
+8. Create a `perf-test-end-time.txt` for [capture_resource_metrics script](https://github.com/integr8ly/integreatly-operator/blob/master/scripts/capture_resource_metrics.sh)
+
+- to get the exact time track the Hyperfoil controller log
+
+9. Collect the data about the performance test run
+
+- from [alerts-during-perf-testing](https://github.com/integr8ly/integreatly-operator/blob/master/scripts/alerts-during-perf-testing.sh)
+- from Hyperfoil
+  - install Hyperfoil locally
+  - bin/cli.sh
+  - connect <hyperfoil-url-without-protocol> -p <port-8090-is-default>
+  - runs # to see all the runs
+  - status <your-run-name>
+  - stats <your-run-name>
+  - export -f json -d . <your-run-name> # to export the data
+  - use [report tool](https://github.com/Hyperfoil/report) to generate the HTML out of the exported data
+- eye review of various Grafana Dashboards, see [this guide](https://docs.google.com/document/d/1KznoB-we73lGUViJApVHyBoIgh3xpgyak6ODAEAHbwk/edit?usp=sharing) on how to do it
+- use [capture_resource_metrics script](https://github.com/integr8ly/integreatly-operator/blob/master/scripts/capture_resource_metrics.sh) to get the data
+- create a spreadsheet with all the collected data
```

Comment:
I wouldn't create a new spreadsheet each time.
I thought that we would just add a new column into existing spreadsheet, so it will be easy to compare to previous runs.
perhaps for easier comparison, we need to create a sheet per request number - one for 50M, one for 20M... Wdyt?

Reply:
This one is tricky. I guess we fully finalize the process only after doing few rounds of testing and result recording. For now I would say let's go with the simplest option, which is just adding the column. I think that here is a way to filter columns out if you want to easily compare just some runs so no need to have a sheet per request number

Reply:
I will update accordingly

Reply:
updated